### ISSUE TITLE
AppImage currently looking for plugins in /tmp/.mount_darktaKdDFDw/usr/share/darktable/lua/contrib/ :(

### ISSUE THREAD

**Issue by vitaly-zdanevich** [darktable-org/darktable#17632]
Hi, would it be better to hardcode the absolute path `/usr/share/darktable/lua/contrib/`?

Because - currently I need to symlink that folder to /tmp on every run?

**Comment by wpferguson** [comment]
How did the lua scripts get there?

**Comment by wpferguson** [comment]
Are you running a nightly appimage build?

**Comment by vitaly-zdanevich** [comment]
> Are you running a nightly appimage build?

https://github.com/darktable-org/darktable/releases/download/release-4.8.1/darktable-4.8.1-x86_64.AppImage

**Comment by vitaly-zdanevich** [comment]
Try to run with `-d lua`

**Comment by wpferguson** [comment]
I downloaded that one and tested with it.  I started the appimage and when the UI appeared `scripts installer` was at the bottom left, because the scripts were not installed.  I installed the scripts using `scripts_installer` and got a message to restart darktable because the downloaded (master) version of the scripts was too new.  I restarted darktable and the scripts worked correctly.

**Comment by vitaly-zdanevich** [comment]
> EDIT 2: /usr/share/darktable/lua is the darktable system lua directory, not the user scripts directory.

Yes, I have installed global scripts to this directory - from system package manager, and AppImage version of Darktable does not see it :(

**Comment by vitaly-zdanevich** [comment]
On start with `-d lua` I see in terminal:

```
3.2184 LUA ERROR : /home/vitaly/.config/darktable/luarc:2: module 'contrib/dtMediaWiki/dtMediaWiki' not found:
	no field package.preload['contrib/dtMediaWiki/dtMediaWiki']
	no file '/usr/local/share/lua/5.4/contrib/dtMediaWiki/dtMediaWiki.lua'
	no file '/usr/local/share/lua/5.4/contrib/dtMediaWiki/dtMediaWiki/init.lua'
	no file '/usr/local/lib/lua/5.4/contrib/dtMediaWiki/dtMediaWiki.lua'
	no file '/usr/local/lib/lua/5.4/contrib/dtMediaWiki/dtMediaWiki/init.lua'
	no file './contrib/dtMediaWiki/dtMediaWiki.lua'
	no file './contrib/dtMediaWiki/dtMediaWiki/init.lua'
	no file '/tmp/.mount_darktaOZyR94/usr/share/darktable/lua/contrib/dtMediaWiki/dtMediaWiki.lua'
	no file '/home/vitaly/.config/darktable/lua/contrib/dtMediaWiki/dtMediaWiki.lua'
	no file '/usr/local/lib/lua/5.4/contrib/dtMediaWiki/dtMediaWiki.so'
	no file '/usr/local/lib/lua/5.4/loadall.so'
	no file './contrib/dtMediaWiki/dtMediaWiki.so'
```

The key line here:

> no file '/tmp/.mount_darktaOZyR94/usr/share/darktable/lua/contrib/dtMediaWiki/dtMediaWiki.lua'

**Comment by wpferguson** [comment]
> Yes, I have installed global scripts to this directory - from system package manager, and AppImage version of Darktable does not see it :(

Because that isn't where they go.  The go in $HOME/.config/darktable/lua

**Comment by vitaly-zdanevich** [comment]
But in what location should a system package manager to install plugins? Only some global folder.

Darktable **not** from AppImage - reads from `/usr/share/darktable/lua/contrib/`

**Comment by vitaly-zdanevich** [comment]
Related link in a system package manager install script

https://github.com/gentoo/guru/blob/016448eb40368bfa9d46c28c2cac5cd42437e7a2/media-plugins/dtmediawiki/dtmediawiki-9999.ebuild#L28

**Comment by wpferguson** [comment]
Now I understand how they got there.

Whoever wrote the install script appears to have not read [the lua-scripts install instructions](https://github.com/darktable-org/lua-scripts#download-and-install).

<install dir>/share/darktable/lua is for the darktable Lua API.  User scripts go in `$HOME/darktable/lua`. 

There is a plan to bundle the lua-scripts with the darktable distribution, but the scripts will be in `<install dir>/share/darktable/lua-scripts`.  There are API changes, as well as script changes, required in order to make that work.

**Comment by vitaly-zdanevich** [comment]
> User scripts go in `$HOME/darktable/lua`

But system package manager install files only to some global folders.

> Whoever wrote the install script

Me, I can fix it.

**Comment by vitaly-zdanevich** [comment]
@stkw0 what do you think?

**Comment by stkw0** [comment]
That you are mixing AppImage and distribution packages, that's obviously problematic. AppImage doesn't mix well with distro packages, so you have to choose between AppImages or install packages from distribution repositories. Then, if there are problems with distribution packages it should be reported first to the distribution.

**Comment by wpferguson** [comment]
If you install the appimage fresh, then start it the lua scripts installer appears in the bottom of the left panel in lighttable.  You can open it, click the execute button, and the lua-scripts will be installed to the correct location (!$HOME/.config/darktable/lua).  You can then add whatever other scripts you desire, either by using script manager or by hand.  You do need to have git installed in order for scripts installer to do it's work.

**Comment by vitaly-zdanevich** [comment]
> Then, if there are problems with distribution packages it should be reported first to the distribution

I reported - cannot build Darktable https://packages.gentoo.org/packages/media-gfx/darktable/bugs

**Comment by wpferguson** [comment]
@vitaly-zdanevich looking back over this I'm wondering if your original question was meant to be "How do I install extra scripts to an appimage darktable installation?"

**Comment by victoryforce** [comment]
AppImage is not installed on the system, instead, at startup it is loop-mounted via FUSE to the /tmp subdirectory. This means that other packages (such as packaged Lua scripts) that expect darktable to be present in the default installation location are incompatible with AppImage.

This is just a fact, not a problem that can somehow be fixed in such packages, or in darktable packaged as an AppImage. Therefore, I propose to close this issue.

**Comment by wpferguson** [comment]
If you run the appimage and have git installed, the scripts installer will install the lua-scripts and they will work fine.

Trying to install lua-scripts to <install dir>/share/darkable/lua will not work regardless of what type of package it is.

I was trying to find out what he was trying to do (I believe install dtMediaWiki.lua) so that I could offer alternatives.

**Comment by vitaly-zdanevich** [comment]
So, please, I do not understand. I want to try AppImage with [dtMediaWiki](https://github.com/trougnouf/dtMediaWiki), it has [a few deps](https://github.com/trougnouf/dtMediaWiki/blob/de2a4dc51304e3319a12dc8b4a80ef41806c3cef/dtmediawiki-dev-1.rockspec#L13-L15) - what is the correct way to copy something/softlink? I am on Gentoo Linux.
